### DISCUSSION TITLE
[Soz-shops and Qb-core] Script error

### DISCUSSION THREAD

**Discussion by Galzerkino** [SOZ-Faut-etre-Sub/SOZ-FiveM-Server#118] [Q&A] [answered]
### Issue

Ok, so i followed your guide to install to avoid problem on installing...
Server start correctly, creation of character good...
i go to any clothing shop and i see 
![image](https://user-images.githubusercontent.com/91811355/219618290-c1a9fecd-0e72-4adc-bede-4def9e4d8db8.png)

![image](https://user-images.githubusercontent.com/91811355/219618326-48617bf1-7e75-41ff-8676-b0e444018b9a.png)

Go to TriggerCallback and found the line 253
![image](https://user-images.githubusercontent.com/91811355/219618239-ead2201e-d5e1-4004-81e9-10a43367fcb7.png)

Go to events.lua
![image](https://user-images.githubusercontent.com/91811355/219618250-cdfb9d9e-fd83-4de9-bbd3-580eebc4e7bc.png)

first error that i got is 
-- GetShopContent
QBCore.Functions.CreateCallback("shops:server:GetShopContent", function(source, cb, shop_name)
    local Player = QBCore.Functions.GetPlayer(source)
    local PlayerModelHash = Player.PlayerData.skin.Model.Hash

    cb(ShopsContent[PlayerModelHash][Shops[shop_name]].items or {})
end)

This is the first error that i got

i don't know how to fix it.

**Comment by Zocaxu** [comment, marked as answer]
1st : you should move to Discussions. This section is reserved to In Game Bug reports, Feature improvements & Feature proposals, not server installation related issues.
2nd : Unfortunately, shop migration is not Open Source yet. The content of the shop is stored in the database of your server which is currently empty when it comes to clothes. This is why you cannot see the content of the shop.
You therefore won't find the same clothes as the ones on SOZ in your shops.
I am somehow working on shops currently. Maybe they will be open source later !

For your interest: the issue you are having is that the server is returning an empty shop content. The code is then trying to index it for reading, but indexing a nil value yields a script error.
